components/patient-measurements/rest/src/main/java/org/phenotips/data/rest/internal/DefaultMeasurementChartResourcesResourceImpl.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.data.rest.internal;
+
+import org.phenotips.data.rest.MeasurementChartResourcesResource;
+import org.phenotips.measurements.MeasurementHandler;
+import org.phenotips.measurements.MeasurementsChartConfiguration;
+import org.phenotips.measurements.internal.MeasurementUtils;
+
+import org.xwiki.bridge.DocumentAccessBridge;
+import org.xwiki.component.annotation.Component;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+/**
+ * Default implementation for {@link MeasurementChartResourcesResource} using XWiki's support for REST resources.
+ *
+ * @version $Id$
+ * @since 1.2M5
+ */
+@Component
+@Named("org.phenotips.data.rest.internal.DefaultMeasurementChartResourcesResourceImpl")
+@Singleton
+public class DefaultMeasurementChartResourcesResourceImpl extends AbstractMeasurementRestResource implements
+        MeasurementChartResourcesResource
+{
+    /** Provides access to the XWiki data. */
+    @Inject
+    private DocumentAccessBridge bridge;
+
+    /** The set of chart resource objects corresponding to this request. */
+    private List<ChartResource> charts;
```

Comment:
I had originally used a class property so that the list generated from the current request could be used in different methods, if necessary. Since the list is reset with every request, this usage is safe as far as I can tell, but I do agree that this isn't the best practice given that it's a singleton with state persisting across requests. I'll go with your recommended change.